platform-tests/src/api/api_test.go

```diff
@@ -0,0 +1,130 @@
+			Team:       "main",
+		}
+		Expect(deployment.AtcAddress).ToNot(Equal(""), "CONCOURSE_ATC_URL environment variable must be set")
+		Expect(deployment.Password).ToNot(Equal(""), "CONCOURSE_ATC_PASSWORD environment variable must be set")
+		Expect(deployment.Username).ToNot(Equal(""), "CONCOURSE_ATC_USERNAME environment variable must be set")
+		Expect(deployment.Version).ToNot(Equal(""), "PIPELINE_TRIGGER_VERSION environment variable must be set")
+
+		monitor := NewMonitor(cfg)
+
+		monitor.Add("Listing all apps in a space", func(cf *cfclient.Client) error {
+			org, err := cf.GetOrgByName("admin")
+			if err != nil {
+				return fmt.Errorf("Failed to fetch 'admin' org: %s", err)
+			}
+
+			space, err := cf.GetSpaceByName("healthchecks", org.Guid)
+			if err != nil {
+				return fmt.Errorf("Failed to fetch 'healthchecks' space within 'admin' org: %s", err)
+			}
+			query := url.Values{}
+			query.Add("q", fmt.Sprintf("organization_guid:%s", org.Guid))
+			query.Add("q", fmt.Sprintf("space_guid:%s", space.Guid))
+			apps, err := cf.ListAppsByQuery(query)
+			if err != nil {
+				return fmt.Errorf("Failed to fetch apps within space 'healthchecks' in org 'admin': %s", err)
+			} else if len(apps) < 1 {
+				return fmt.Errorf("expected at least one app in the 'healthchecks' space got %d", len(apps))
+			}
+
+			return nil
+		})
+
+		monitor.Add("Fetching detailed app information", func(cf *cfclient.Client) error {
+			org, err := cf.GetOrgByName("admin")
+			if err != nil {
+				return fmt.Errorf("Failed to fetch 'admin' org")
+			}
+
+			apps, err := cf.ListAppsByQuery(url.Values{"q": []string{
+				"name:" + appName,
+				"organization_guid:" + org.Guid,
+			}})
+			if err != nil {
+				return err
+			} else if len(apps) != 1 {
+				return fmt.Errorf("expected a single app named '%s' got %d", appName, len(apps))
+			}
+			app := apps[0]
+
+			if _, err := cf.GetAppStats(app.Guid); err != nil {
+				return fmt.Errorf("Failed to fetch app stats: %s", err)
+			}
+
+			if _, err := cf.GetAppInstances(app.Guid); err != nil {
+				return fmt.Errorf("Failed to fetch app instances: %s", err)
+			}
+
+			if _, err := cf.GetAppRoutes(app.Guid); err != nil {
+				return fmt.Errorf("Failed to fetch app routes: %s", err)
+			}
+
+			return nil
+		})
+
+		// poll concourse job status til done
+		go func(concourseConnectionAttemptsRemaining int64) {
+			defer GinkgoRecover()
+			for {
+				<-time.After(2 * time.Second)
+				if done, err := deployment.Complete(); err != nil {
+					concourseConnectionAttemptsRemaining--
+					if concourseConnectionAttemptsRemaining <= 0 {
+						monitor.Stop()
+						Expect(err).ToNot(HaveOccurred())
+						return
+					}
+					lg("failed to get status from concourse [", concourseConnectionAttemptsRemaining, " attempts remaining]", err)
+				} else if done {
+					concourseConnectionAttemptsRemaining = 5
+					lg("detected deployment job completed, stopping monitor")
+					monitor.Stop()
+					return
+				}
+			}
+		}(5)
```

Comment:
Could we state that `5` and the one from line 115 in a constant for easier management in the future?